File: Backend/model/Rule.java
package com.realestate.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "rules")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Rule {

    public enum Role {
        MANAGER,
        ADMIN,
        BUYER,
        SELLER
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String code;  // Unique rule code or identifier

    @Column(nullable = false)
    private String description;  // The rule description

    @Column(nullable = false)
    private boolean active;  // Is the rule active or not

    @Enumerated(EnumType.STRING)
    private Role targetRole;  // Role this rule applies to
}

File: Frontend/admin/model/rule.ts

export enum Role {
  ALL = 'ALL',
  MANAGER = 'MANAGER',
  ADMIN = 'ADMIN',
  BUYER = 'BUYER',
  SELLER = 'SELLER'
}

  
  export interface Rule {
    id: number;
    code: string;
    description: string;
    active: boolean;
    targetRole: Role;
  }
  